go/cmd/vtctldclient/command/keyspaces.go

```diff
@@ -503,8 +449,6 @@ func commandValidateVersionKeyspace(cmd *cobra.Command, args []string) error {
 func init() {
 	CreateKeyspace.Flags().BoolVarP(&createKeyspaceOptions.Force, "force", "f", false, "Proceeds even if the keyspace already exists. Does not overwrite the existing keyspace record")
 	CreateKeyspace.Flags().BoolVarP(&createKeyspaceOptions.AllowEmptyVSchema, "allow-empty-vschema", "e", false, "Allows a new keyspace to have no vschema")
-	CreateKeyspace.Flags().StringVar(&createKeyspaceOptions.ShardingColumnName, "sharding-column-name", "", "The column name to use for sharding operations")
-	CreateKeyspace.Flags().Var(&createKeyspaceOptions.ShardingColumnType, "sharding-column-type", "The type of the column to use for sharding operations")
```

Comment:
@ajm188 what is the right way to deprecate these options?

Reply:
[`pflag` has native support for this](https://pkg.go.dev/github.com/spf13/pflag#readme-deprecating-a-flag-or-its-shorthand)!! (_vaguely gestures at "won't it be great to use `pflag` everywhere in the future?"_):

```go
CreateKeyspace.Flags().StringVar(&createKeyspaceOptions.ShardingColumnName, "sharding-column-name", ...)
CreateKeyspace.Flags().MarkDeprecated("sharding-column-name", "Keyspace.ShardingColumnName is no longer used for resharding") // feel free to tweak the deprecation message, i didn't give this much thought
// same pattern for ShardingColumnType
```

Reply:
Thank you. I'm thinking we should let this PR delete these flags on main (to go into v15) and do a separate PR to deprecate them in v14 (on release-14.0)

Reply:
makes sense to me!